docs/root/configuration/http/http_filters/grpc_json_reverse_transcoder_filter.rst

```diff
@@ -0,0 +1,55 @@
+.. _config_http_filters_grpc_json_reverse_transcoder:
+
+gRPC-JSON reverse transcoder
+============================
+
+* gRPC :ref:`architecture overview <arch_overview_grpc>`
+* This filter should be configured with the type URL ``type.googleapis.com/envoy.extensions.filters.http.grpc_json_reverse_transcoder.v3.GrpcJsonReverseTranscoder``.
+* :ref:`v3 API reference <envoy_v3_api_msg_extensions.filters.http.grpc_json_reverse_transcoder.v3.GrpcJsonReverseTranscoder>`
+
+This is a filter which allows a gRPC client to send requests to Envoy and get proxied
+to a RESTful JSON API service. The HTTP mapping for the gRPC service has to be defined by
+`custom options <https://cloud.google.com/service-management/reference/rpc/google.api#http>`_.
+
+JSON mapping
+------------
+
+The protobuf to JSON mapping is defined `here <https://developers.google.com/protocol-buffers/docs/proto3#json>`_.
+
+.. _config_grpc_json_reverse_transcoder_generate_proto_descriptor_set:
+
+How to generate proto descriptor set
+------------------------------------
+
+Envoy has to know the proto descriptor of your gRPC service in order to do the transcoding.
+
+To generate a protobuf descriptor set for the gRPC service, you'll also need to clone the
```

Comment:
This should use the googleapis that are provided by Envoy's bazel setup - same with protoc - its provided by Envoys build system

Reply:
As far as I know `protoc` looks for the actual proto files and not the bazel targets, so how can that be used with `protoc`, CMIIAW.

Reply:
im struggling to find a good (standalone) example

a basic example usage of the envoy configured `protoc`:

```console
$ bazel run @com_google_protobuf//:protoc -- --version
```
this is an example of `run` usage - to use it together with the envoy supplied googleapis dep you kinda need to figure out a `build` pattern

the googleapis are available from `@com_google_googleapis` - if you grep the envoy code for that string you will see the various ways these are used

the complexity is that i guess you want to combine `protoc` + `googleapis` + user code - in that case you would need to add a `build` target with a configurable input

trying to think if there are any existing patterns to copy - probably there is, but cant think of any right now